control-plane/agents/core/src/volume/scheduling.rs

```diff
@@ -0,0 +1,33 @@
+use crate::core::{
+    registry::Registry,
+    scheduling::{
+        resources::ReplicaItem,
+        volume,
+        volume::{GetChildForRemoval, GetSuitablePools},
+        ResourceFilter,
+    },
+    wrapper::PoolWrapper,
+};
+
+/// Return a list of pre sorted pools to be used by a volume
+pub(crate) async fn get_volume_pool_candidates(
+    request: impl Into<GetSuitablePools>,
+    registry: &Registry,
+) -> Vec<PoolWrapper> {
+    volume::IncreaseVolumeReplica::builder_with_defaults(request, registry)
+        .await
+        .collect()
+        .into_iter()
+        .map(|e| e.collect())
+        .collect()
+}
+
+/// Select a nexus child to be removed from a nexus
```

Comment:
It looks like we are returning a list of candidates rather than a particular child. Could we change this comment to be similar to line 12 as that was really clear?

Reply:
Done